.github/workflows/build_and_release_ecr_image.yml

```
@@ -0,0 +1,123 @@
+name: CI & CD
+
+on: [push]
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  lint-and-test:
+    name: 'Lint & Test'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```

```suggestion
        uses: actions/checkout@v4
```
That's the up-to-date action :)